documentation/modules/new-migrating-virtual-machines-cli.adoc

```diff
@@ -0,0 +1,1036 @@
+====
+
+ifdef::vmware[]
+[start=7]
+. Create a `Plan` manifest for the migration:
++
+[source,yaml,subs="attributes+"]
+----
+$ cat << EOF | {oc} apply -f -
+apiVersion: forklift.konveyor.io/v1beta1
+kind: Plan
+metadata:
+  name: <plan> <1>
+  namespace: <namespace>
+spec:
+  warm: false <2>
+  provider:
+    source:
+      name: <source_provider>
+      namespace: <namespace>
+    destination:
+      name: <destination_provider>
+      namespace: <namespace>
+  map: <3>
+    network: <4>
+      name: <network_map> <5>
+      namespace: <namespace>
+    storage: <6>
+      name: <storage_map> <7>
+      namespace: <namespace>
+  targetNamespace: <target_namespace>
+  vms: <8>
+    - id: <source_vm> <9>
+    - name: <source_vm>
+      hooks: <10>
+        - hook:
+            namespace: <namespace>
+            name: <hook> <11>
+          step: <step> <12>
+EOF
+----
+<1> Specify the name of the `Plan` CR.
+<2> Specify whether the migration is warm - `true` - or cold - `false`. If you specify a warm migration without specifying a value for the `cutover` parameter in the `Migration` manifest, only the precopy stage will run.
+<3> Specify only one network map and one storage map per plan.
+<4> Specify a network mapping even if the VMs to be migrated are not assigned to a network. The mapping can be empty in this case.
+<5> Specify the name of the `NetworkMap` CR.
+<6> Specify a storage mapping even if the VMs to be migrated are not assigned with disk images. The mapping can be empty in this case.
+<7> Specify the name of the `StorageMap` CR.
+<8> You can use either the `id` _or_ the `name` parameter to specify the source VMs. +
+<9> Specify the VMware vSphere VM MOR..
+<10> Optional: You can specify up to two hooks for a VM. Each hook must run during a separate migration step.
+<11> Specify the name of the `Hook` CR.
+<12> Allowed values are `PreHook`, before the migration plan starts, or `PostHook`, after the migration is complete.
+endif::[]
+
+ifdef::rhv[]
+[start=6]
+. Create a `Plan` manifest for the migration:
++
+[source,yaml,subs="attributes+"]
+----
+$ cat << EOF | {oc} apply -f -
+apiVersion: forklift.konveyor.io/v1beta1
+kind: Plan
+metadata:
+  name: <plan> <1>
+  namespace: <namespace>
+  preserveClusterCpuModel: true <2>
+spec:
+  warm: true <3>
+  provider:
+    source:
+      name: <source_provider>
+      namespace: <namespace>
+    destination:
+      name: <destination_provider>
+      namespace: <namespace>
+  map: <4>
+    network: <5>
+      name: <network_map> <6>
+      namespace: <namespace>
+    storage: <7>
+      name: <storage_map> <8>
+      namespace: <namespace>
+  targetNamespace: <target_namespace>
+  vms: <9>
+    - id: <source_vm> <10>
+    - name: <source_vm>
+      hooks: <11>
+        - hook:
+            namespace: <namespace>
+            name: <hook> <12>
+          step: <step> <13>
+EOF
+----
+<1> Specify the name of the `Plan` CR.
+<2> See note below.
```

Comment:
@ahadas I didn't upload the note (include it in the file) because I am having a formatting problem with it -- but if you're good with the text, I'll go back to working on its formatting.

Reply:
Comment added.